### PR TITLE
[FEATURE#26106] Upgrade du provider

### PR DESCRIPTION
**Pull request by gdubost**
Passage a Guzzle 6 !

### CODE REVIEW

**Comment by TinyTinyKame** [comment]
Okay

File: src/ConversationManager.php [hunk 1]
```diff
@@ -104,15 +97,25 @@ public function toJsonResponse(array $conversations)
         return $conversations;
     }
 
-    private function fireRequest(GuzzleRequest $request, $cookie)
+    private function fireRequest($method, $uri, $body = [])
     {
+        $method = strtoupper($method);
+
+        if (false === in_array($method, ["GET", "POST", "PUT", "DELETE", "OPTIONS"])) {
+            throw new Exception("ConversationProxy can not fire request of method : {$method}", 500);
```
Comment (faouzic) [inline src/ConversationManager.php]: 405 method not allowed

File: src/DumbMethodsProxy.php [hunk 2]
```diff
@@ -46,108 +46,85 @@ public function connect(Application $app)
...
+        return $this->fireRequest($app, $req);
     }
 
-    public function doProxyWithoutDatas(Application $app, Request $req)
+    public function fireRequest(Application $app, Request $req, $remove_prefix = "")
     {
         $method = $req->getMethod();
         if (false === in_array($method, ["GET", "POST", "PUT", "DELETE", "OPTIONS"])) {
-            throw new Exception("ConversationProxy can not fire request of method : {$method}");
+            throw new Exception("ConversationProxy can not fire request of method : {$method}", 500);
```
Comment (faouzic) [inline src/DumbMethodsProxy.php]: same

**Comment by faouzic** [comment]
Ok

**Comment by lapwil** [comment]
OK

**Pull request merged**